pkg/liqoctl/status/peer/peerinfo.go

```diff
@@ -0,0 +1,352 @@
+// holds the information about the peered cluster.
+type PeerInfoChecker struct {
+	o                *status.Options
+	pio              *PeerInfoOptions
+	peerInfoSection  output.Section
+	collectionErrors []status.CollectionError
+}
+
+const (
+	peerInfoCheckerName = "Peered Cluster Information"
+)
+
+// NewPeerInfoChecker return a new PeerInfoChecker.
+func NewPeerInfoChecker(o *status.Options, remoteClusterNames ...string) *PeerInfoChecker {
+	return &PeerInfoChecker{
+		peerInfoSection: output.NewRootSection(),
+		o:               o,
+		pio: &PeerInfoOptions{
+			RemoteClusterNames: remoteClusterNames,
+		},
+	}
+}
+
+// Silent implements the Check interface.
+func (pic *PeerInfoChecker) Silent() bool {
+	return false
+}
+
+// Collect implements the collect method of the Checker interface.
+// it collects the infos of the peered cluster.
+func (pic *PeerInfoChecker) Collect(ctx context.Context) {
+	localClusterIdentity, err := liqoutils.GetClusterIdentityWithControllerClient(ctx, pic.o.CRClient, pic.o.LiqoNamespace)
+	localClusterName := localClusterIdentity.ClusterName
+	if err != nil {
+		pic.addCollectionError("LocalClusterName", "", err)
+		return
+	}
+
+	foreignClusterList, err := liqogetters.ListForeignClustersByLabel(ctx, pic.o.CRClient, labels.Everything())
+	if err != nil {
+		pic.addCollectionError("ForeignClusters", "", err)
+		return
+	}
+
+	remoteClusterNamesNotFound := pic.checkPeerExistence(foreignClusterList)
+
+	foreignClusterListSelected := pic.getForeignClusterListSelected(foreignClusterList)
+
+	for i := range foreignClusterListSelected.Items {
+		fc := &foreignClusterListSelected.Items[i]
+		remoteClusterName := fc.Spec.ClusterIdentity.ClusterName
+		if slices.Contains(remoteClusterNamesNotFound, remoteClusterName) {
+			pic.peerInfoSection.AddSectionWithDetail(remoteClusterName, PeerNotFoundMsg)
+			continue
+		}
+
+		remoteClusterID := fc.Spec.ClusterIdentity.ClusterID
+
+		clusterSection := pic.peerInfoSection.AddSectionWithDetail(remoteClusterName, remoteClusterID)
+
+		pic.addPeerSection(clusterSection, fc)
+
+		pic.addAuthSection(clusterSection, fc)
+
+		err = pic.addNetworkSection(ctx, clusterSection, fc, localClusterName)
+		if err != nil {
+			pic.addCollectionError("Network", "unable to collect Network", err)
+		}
+
+		err = pic.addResourceSection(ctx, clusterSection, remoteClusterID, localClusterName, remoteClusterName)
+		if err != nil {
+			pic.addCollectionError("Resources", "unable to collect Resources", err)
+		}
+	}
+}
+
+// getForeignClusterList returns the list of ForeignCluster objects.
+func (pic *PeerInfoChecker) getForeignClusterListSelected(
+	foreignClusterList *discoveryv1alpha1.ForeignClusterList) *discoveryv1alpha1.ForeignClusterList {
+	foreignClusterListSelected := discoveryv1alpha1.ForeignClusterList{}
+	if len(pic.pio.RemoteClusterNames) == 0 {
+		foreignClusterListSelected = *foreignClusterList
+	} else {
+		for _, rcn := range pic.pio.RemoteClusterNames {
```

Comment:
I would suggest using a map to simplify this code (and drop the function below).